UPGRADE.md

```diff
@@ -797,13 +797,53 @@ const OrderEdit = (props) => (
 
 ## Reference inputs used to inject an helperText inside meta
 
-`<ReferenceInput>` used to inject an `helperText` inside `meta` when they were unable to retrieve referenced records.
+`<ReferenceInput>` used to send a `helperText` inside the `meta` prop of its child whenever it was unable to retrieve referenced records. The same goes for `<ReferenceArrayInput>`.
 
-They now set this issue as an error inside the meta just like final-form does and won't override their child `helperText` anymore.
+These components now use the `meta: { error }` prop and won't override their child `helperText` anymore.
 
 Furthermore, they don't support the `helperText` at all anymore as this is a pure UI concern which should be handled by their child.
 
+If you've implemented a custom child component for `<ReferenceInput>` of `<ReferenceArrayInput>`, you must now display the `error` in your component.
+
+```diff
+const MySelectIpnut = ({
+    // ...
+-   meta: { helperText }
++   meta: { error }
+}) => (
+    if (error) {
```

Comment:
I'm not sure about this sample. The error is inside the meta, just like any final-form error. Custom components should display it like any other form error